Action and launchers part 2 (#185)
elementary/docs: apis/launchers.md

Actions are specific functions your app can perform without already being open; think of them as alternate and more specific entry points into your app. Actions appear in the context menu of your app icon in the Applications Menu and Dock, and are searchable by name from the Applications Menu.

### D-Bus activation

> Your app needs to support D-Bus activation in order to use actions as entry points. This does not require any changes to the application source code. All that is needed is a service file which is not unlike the `.desktop` file that you are already familiar with. Define a new `service.in` file in the `data` directory:
> 
> ```ini
> [D-BUS Service]
> Name=com.github.myteam.myapp
> Exec=@exec_name@ --gapplication-service
> ```
> 
> Notice the `@exec_name@` configurable variable and the `.in` file extension. Here we are using meson's [configuration feature](https://mesonbuild.com/Configuration.html) to specify the name of the executable. To install the service add this to your `meson.build` file:
> 
> ```coffeescript
> # Install D-Bus service, so that application can be started by D-Bus
> service_conf_data = configuration_data()
> service_conf_data.set('exec_name', get_option('prefix') / get_option('bindir') / meson.project_name())
> 
> configure_file(
>     input: 'data' / 'service.in',
>     output: meson.project_name() + '.service',
>     configuration: service_conf_data,
>     install: true,
>     install_dir: get_option('datadir') / 'dbus-1' / 'services'
> )
> ```

Reviewer: We can simplify this quite a bit by just putting in the exec name ourselves. That would be I think better for a tutorial
```suggestion
Your app needs to support D-Bus activation in order to use actions as entry points. This does not require any changes to the application source code. All that is needed is a service file which is not unlike the `.desktop` file that you are already familiar with. Create a new `.service` file in the `data` directory:

```ini
[D-BUS Service]
Name=com.github.myteam.myapp
Exec=com.github.myteam.myapp --gapplication-service
```

To install the service add the following to your `meson.build` file:

```coffeescript
# Install D-Bus service, so that application can be started by D-Bus
install_data(
    'data' / 'myapp.service',
    install_dir: get_option('datadir') / 'dbus-1' / 'services',
    rename: meson.project_name() + '.service',
)
```

Author: I wrongly thought we needed the `prefix/bindir` in there, but the non-template version is much better!

```ini
[Desktop Entry]
Name=Hello Again
[...]
Actions=my-action;
```

Then use a dedicated group, named after the unique action name, to define the details of each action:

```ini
[Desktop Action my-action]
Name=My Great Action
Icon=com.github.myteam.myapp.my-action-icon
Exec=com.github.myteam.myapp
```

{% hint style="warning" %}
The action name used in `.desktop` file, both in Desktop Entry and later in Desktop Action groups, needs to match exactly the name used to register the action with `GLib.Applicatin` in the source code.
{% endhint %}

The `Icon` line is optional and should be an icon which represents the action that will be performed. The `Exec` line should be specified, but is used only for backwards compatibility in case your app ever runs in an environment without D-Bus activation support.

{% hint style="info" %}
The action name should not include your app's name, as it will always be displayed alongside your app. The action icon should also not be your app icon, as it may be shown in the menu for your app icon, or badged on top of the app icon.
{% endhint %}

> 

Reviewer: Can we add a link in here to the code sample like we do in the Actions section?

Author: Yes! I [added the link](https://github.com/elementary/docs/pull/185/commits/20d69b1afe41af0745fb26d7f287360190b61f4a) to the bottom of the page. I also flattened this text here about action names and icons from info box into a standard paragraph. I was afraid that having every second paragraph be styled as info or warning would be distracting.
